src/prefect/client/schemas/objects.py

```diff
@@ -156,6 +158,23 @@ class StateDetails(PrefectBaseModel):
     task_parameters_id: Optional[UUID] = None
 
 
+def is_none_or_data(v, handler, info):
+    if v is None:
+        return v
+
+    try:
+        return handler(v, info)
+    except TypeError as e:
+        # allow backwards compatibility, i.e. `Completed(data={"A": 1})`
+        if "Can't instantiate abstract class" in str(e):
```

Comment:
it feels like we should be able to change `BaseResult` to avoid this entirely, but I haven't figured out how to do that yet